Basic-Car-Maintenance/Shared/AppIntents/AddOdometerReadingAppIntent.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  AddOdometerReadingAppIntent.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Omar Hegazy on 23/08/2024.
+//
+
+import Foundation
+import AppIntents
+
+/// The query used to retrieve vehicles for adding odometer.
+struct VehicleQuery: EntityQuery {
+   
+    func entities(for identifiers: [Vehicle.ID]) async throws -> [Vehicle] {
+        try await fetchVehicles()
+    }
+    
+    func suggestedEntities() async throws -> [Vehicle] {
+        try await fetchVehicles()
+    }
+    
+    private func fetchVehicles() async throws -> [Vehicle] {
+        let authViewModel = await AuthenticationViewModel()
+        let odometerVM = OdometerViewModel(userUID: authViewModel.user?.uid)
```

Comment:
So from what I've seen in examples we'd use `@Dependency` for these dependencies in the `EntityQuery` and `AppIntent`

Like:
`@Dependency private var authViewModel: AuthenticationViewModel` as a struct level variable, then we use it in perform and `fetchVehicles`, rather than initializing it

Can you add that here and in the other places too

Reply:
@OmarHegazy93